sdk/tables/azure-data-tables/inc/azure/data/tables/table_client.hpp

```diff
@@ -131,18 +131,26 @@ namespace Azure { namespace Data { namespace Tables {
         Core::Context const& context = {});
 
     /**
-     * @brief Upsert specified entity in a table.
+     * @brief Update or insert specified entity in a table.
      *
      * @param tableEntity The TableEntity to upsert.
-     * @param upsertKind Upsert kind(update/merge), default update.
      * @param context for canceling long running operations.
      * @return Upsert entity result.
      */
-    Response<Models::UpsertEntityResult> UpsertEntity(
+    Response<Models::UpdateEntityResult> UpdateInsertEntity(
         Models::TableEntity const& tableEntity,
-        Models::UpsertKind upsertKind = Models::UpsertKind::Update,
         Core::Context const& context = {});
 
+    /**
+     * @brief Merge or Insert the specified entity in a table.
```

Comment:
consistency nit: We have `Update or insert` on the above method (lower-case). Either is fine as long as we are consistent.
```suggestion
     * @brief Merge or insert the specified entity in a table.
```